core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -19,18 +19,31 @@ package org.apache.spark.metrics
 import java.lang.management.{BufferPoolMXBean, ManagementFactory}
 import javax.management.ObjectName
 
+import scala.collection.mutable
+
+import org.apache.spark.executor.ProcfsBasedSystems
 import org.apache.spark.memory.MemoryManager
 
 /**
  * Executor metric types for executor-level metrics stored in ExecutorMetrics.
  */
 sealed trait ExecutorMetricType {
-  private[spark] def getMetricValue(memoryManager: MemoryManager): Long
-  private[spark] val name = getClass().getName().stripSuffix("$").split("""\.""").last
+  private[spark] def getMetricValue(memoryManager: MemoryManager): Long = 0
+  private[spark] def getMetricSet(memoryManager: MemoryManager): Array[Long] = {
+    new Array[Long](0)
```

Comment:
`Set` isn't a great name for this, as the order matters.  how about `getMetricValues()`?